client/src/containers/SearchPage/index.js

```diff
@@ -0,0 +1,546 @@
+        displayProject: collectProjectData,
+        listProjectData: collectProjectData,
+      });
+    } catch (err) {
+      this.handleError(err);
+    }
+  }
+
+  cohortLocation = (value) => {
+    const { allCohortData, allStudentData, showProjectSection } = this.state;
+    if (value !== 'All') {
+      const fillterdCohort = allCohortData.filter(
+        (e) => e.name.split('')[0].toLowerCase() === value.toLowerCase()
+      );
+
+      const studentForSpecificLocation = [];
+
+      this.getAllStudentOfLocation(
+        fillterdCohort,
+        allStudentData,
+        studentForSpecificLocation
+      );
+
+      const filterCohortPagination = this.setPagination(fillterdCohort);
+
+      if (!showProjectSection) {
+        this.setState({
+          listCohortData: fillterdCohort,
+          displayCohortData: fillterdCohort,
+          listStudentData: studentForSpecificLocation,
+          displayStudent: [],
+          showCohorts: true,
+          ...filterCohortPagination,
+        });
+      }
+    } else if (!showProjectSection) {
+      const allCohortPagination = this.setPagination(allCohortData);
+      this.setState({
+        listCohortData: allCohortData,
+        displayCohortData: allCohortData,
+        listStudentData: allStudentData,
+        displayStudent: [],
+        showCohorts: true,
+        ...allCohortPagination,
+      });
+    }
+  };
+
+  getAllStudentOfLocation = (
+    fillterdCohort,
+    allStudentData,
+    studentForSpecificLocation
+  ) => {
+    fillterdCohort
+      .map((cohort) =>
+        allStudentData.filter((student) => student.cohort_id === cohort.id)
+      )
+      .forEach((arr) =>
+        arr.forEach((student) => studentForSpecificLocation.push(student))
+      );
+  };
+
+  cohortName = (value) => {
+    const { listCohortData, allStudentData, showProjectSection } = this.state;
+    const fillterdCohort = listCohortData.filter((e) => e.id === value);
+    const fillterdStudent = allStudentData.filter(
+      (e) => e.cohort_id === fillterdCohort[0].id
+    );
+
+    if (!showProjectSection) {
+      const filterStudentPagination = this.setPagination(fillterdStudent);
+      this.setState({
+        displayCohortData: fillterdCohort,
+        listStudentData: fillterdStudent,
+        displayStudent: fillterdStudent,
+        showCohorts: false,
+        ...filterStudentPagination,
+      });
+    }
+  };
+
+  studentName = (value) => {
+    const { allStudentData, showProjectSection } = this.state;
+    const studentSelected = allStudentData.filter((e) => e.id === value);
+    if (!showProjectSection) {
+      this.setState({ total: 1, pageNumber: 1 });
+    }
+    this.setState({
+      displayStudent: studentSelected,
+      showCohorts: false,
+      studentId: studentSelected[0].id,
+    });
+  };
+
+  projectName = (value) => {
+    const { allProjectData, showProjectSection } = this.state;
+    const ProjectSelected = allProjectData.filter((e) => e.id === value);
```

Comment:
It will be better to use a more descriptive variable name and do it for the same cases in this file
```suggestion
    const ProjectSelected = allProjectData.filter((project) => e.id === value);
```